src/main/java/com/celements/course/service/CourseService.java

```diff
@@ -316,6 +316,38 @@ public boolean validateParticipant(DocumentReference regDocRef, String emailAdr,
     return success;
   }
 
+  @Override
+  public CourseConfirmState getConfirmeState(DocumentReference objDocRef) {
+    DocumentReference courseParticipantClassRef = getCourseClasses().getCourseParticipantClassRef(
+        modelContext.getXWikiContext().getDatabase());
+    CourseConfirmState confirmState = CourseConfirmState.UNCONFIRMED;
+    try {
+      List<BaseObject> partiObjs = modelAccess.getXObjects(objDocRef, courseParticipantClassRef);
+      int index = 0;
+      for (BaseObject obj : partiObjs) {
+        String state = obj.getStringValue("status");
+        if ((state.equals(CourseConfirmState.CONFIRMED.id) && (index == 0)) || (state.equals(
```

Comment:
after changing `state` to a `CourseConfirmState` you may compare directly with == and !=